### DISCUSSION TITLE
Documentation on how to create subscriptions that never expire

### DISCUSSION THREAD

**Discussion by jackson-stone-ableto** [googleapis/nodejs-pubsub#1314] [Q&A]
I have a use case that requires the creation of subscriptions that never expire. The argument to pass in the docs for the CLI seems to be the string "never". In the typescript project I have, providing "never" is against the type specification for the expirationPolicy option for subscriptions.

I'm allowed to set it to "null" - but I do not know if this is in fact creating it without an expiration (setting it to "never") or is causing it to go to the default value (30 days)

Any help or documentation around this would be awesome!

**Comment by feywind** [comment]
@jackson-stone-ableto For the gcloud, side, I'm assuming you mean this parameter?

```
--expiration-period=EXPIRATION_PERIOD
        The subscription will expire if it is inactive for the given period.
        Valid values are strings of the form INTEGER[UNIT], where UNIT is one
        of "s", "m", "h", and "d" for seconds, minutes, hours, and days,
        respectively. If the unit is omitted, seconds is assumed. This flag
        additionally accepts the special value "never" to indicate that the
        subscription will never expire.
```

which probably maps to this definition in the gapic protos?

```
                /** Subscription expirationPolicy */
                expirationPolicy?: (google.pubsub.v1.IExpirationPolicy|null);
```

It is sort of odd to me that `never` is something we encourage in gcloud, but there doesn't seem to be an equivalent here. That might be an @alexander-fenster question? (Given the difference between the generated types and the protos.)

Based on this:

> ttl | google.protobuf.Duration | Specifies the "time-to-live" duration for an associated resource. The resource expires if it is not active for a period of ttl. The definition of "activity" depends on the type of the associated resource. The minimum and maximum allowed values for ttl depend on the type of the associated resource, as well. If ttl is not set, the associated resource never expires.

...I'm going to guess the right answer is to set `expirationPolicy` to `{}`, but I'll see if Alex has other thoughts.

**Comment by jackson-stone-ableto** [comment]
Thank you @feywind  - You are correct in assessing what I was referring to. I look forward to an answer! :D
